pkg/printer/printer.go

```diff
@@ -384,6 +384,170 @@ func (p *Printer) WriteProtoNodeStatusEvent(r *observerpb.GetFlowsResponse) erro
+		}
+	case pb.AgentEventType_POLICY_UPDATED, pb.AgentEventType_POLICY_DELETED:
+		if p := e.GetPolicyUpdate(); p != nil {
+			return fmt.Sprintf("labels: [%s], revision: %d, count: %d",
+				strings.Join(p.Labels, ","), p.Revision, p.RuleCount)
+		}
+	case pb.AgentEventType_ENDPOINT_REGENERATE_SUCCESS, pb.AgentEventType_ENDPOINT_REGENERATE_FAILURE:
+		if r := e.GetEndpointRegenerate(); r != nil {
+			var sb strings.Builder
+			fmt.Fprintf(&sb, "id: %d, labels: [%s]", r.Id, strings.Join(r.Labels, ","))
+			if re := r.Error; re != "" {
+				fmt.Fprintf(&sb, ", error: %s", re)
+			}
+			return sb.String()
+		}
+	case pb.AgentEventType_ENDPOINT_CREATED, pb.AgentEventType_ENDPOINT_DELETED:
+		if ep := e.GetEndpointUpdate(); ep != nil {
+			var sb strings.Builder
+			fmt.Fprintf(&sb, "id: %d", ep.Id)
+			if n := ep.PodName; n != "" {
+				fmt.Fprintf(&sb, ", pod name: %s", n)
+			}
+			if n := ep.Namespace; n != "" {
+				fmt.Fprintf(&sb, ", namespace: %s", n)
+			}
+			return sb.String()
+		}
+	case pb.AgentEventType_IPCACHE_UPSERTED, pb.AgentEventType_IPCACHE_DELETED:
+		if i := e.GetIpcacheUpdate(); i != nil {
+			var sb strings.Builder
+			fmt.Fprintf(&sb, "cidr: %s, identity: %d", i.Cidr, i.Identity)
+			if i.OldIdentity != nil {
+				fmt.Fprintf(&sb, ", old identity: %d", i.OldIdentity.Value)
+			}
+			if i.HostIp != "" {
+				fmt.Fprintf(&sb, ", host ip: %s", i.HostIp)
+			}
+			if i.OldHostIp != "" {
+				fmt.Fprintf(&sb, ", old host ip: %s", i.OldHostIp)
+			}
+			fmt.Fprintf(&sb, ", encrypt key: %d", i.EncryptKey)
+			return sb.String()
+		}
+	case pb.AgentEventType_SERVICE_UPSERTED:
+		if svc := e.GetServiceUpsert(); svc != nil {
+			var sb strings.Builder
+			fmt.Fprintf(&sb, "id: %d", svc.Id)
+			if fe := svc.FrontendAddress; fe != nil {
+				fmt.Fprintf(&sb, ", frontend: %s", formatServiceAddr(fe))
+			}
+			if bes := svc.BackendAddresses; len(bes) != 0 {
+				backends := make([]string, 0, len(bes))
+				for _, a := range bes {
+					backends = append(backends, formatServiceAddr(a))
+				}
+				fmt.Fprintf(&sb, ", backends: [%s]", strings.Join(backends, ","))
+			}
+			if t := svc.Type; t != "" {
+				fmt.Fprintf(&sb, ", type: %s", t)
+			}
+			if tp := svc.TrafficPolicy; tp != "" {
+				fmt.Fprintf(&sb, ", traffic policy: %s", tp)
+			}
+			if n := svc.Name; n != "" {
+				fmt.Fprintf(&sb, ", name: %s", n)
+			}
+			if ns := svc.Namespace; ns != "" {
+				fmt.Fprintf(&sb, ", namespace: %s", ns)
+			}
+			return sb.String()
+		}
+	case pb.AgentEventType_SERVICE_DELETED:
+		if s := e.GetServiceDelete(); s != nil {
+			return fmt.Sprintf("id: %d", s.Id)
+		}
+	}
+	return "UNKNOWN"
+}
+
+// WriteProtoAgentEvent writes v1.AgentEvent into the output writer.
+func (p *Printer) WriteProtoAgentEvent(r *observerpb.GetFlowsResponse) error {
+	e := r.GetAgentEvent()
+	if e == nil {
+		return errors.New("not an agent event")
+	}
+
+	switch p.opts.output {
+	case JSONOutput:
+		return p.jsonEncoder.Encode(e)
```

Comment:
We're not honoring `p.opts.nodeName` here:
```
% ./hubble --server localhost:4245 observe --debug --print-node-name -o json -t agent | jq
{
  "type": "IPCACHE_UPSERTED",
  "ipcache_update": {
    "cidr": "0.0.0.0/0",
    "identity": 2,
    "old_identity": 2
  }
}
...
```
Interestingly, we don't have this issue with flows as we have [a `node_name` field in the Flow protobuf message](https://github.com/cilium/cilium/blob/44c122d930bafffd95805c3b91a3831ceecf9bfd/api/v1/flow/flow.proto#L46-L47). We also don't have this issue with the jsonpb output because there is a `node_name` field in the GetFlowsResponse message.

I think it's fine as-is though, but maybe we want to document that the `--print-node-name` flag has no effect on `json` and `jsonpb` output (which is the status quo).

Reply:
The idea was that the node name is always present in `json` and `jsonpb` output.